src/main/java/org/kohsuke/github/GHIssueQueryBuilder.java

```diff
@@ -0,0 +1,163 @@
+package org.kohsuke.github;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GHIssueQueryBuilder extends GHQueryBuilder<GHIssue> {
+    private final GHRepository repo;
+    private List<String> labels = new ArrayList<>();
+
+    GHIssueQueryBuilder(GHRepository repo) {
+        super(repo.root());
+        this.repo = repo;
+    }
```

Comment:
This query builder is specifically querying [issues in a repository](https://docs.github.com/en/rest/reference/issues#list-repository-issues). 

It is also possible to query [issues related to the current user](https://docs.github.com/en/rest/reference/issues#list-issues-assigned-to-the-authenticated-user) or [issues in an organization related to the current user)](https://docs.github.com/en/rest/reference/issues#list-organization-issues-assigned-to-the-authenticated-user).  Even more fun, each of those endpoints supports a different set of query parameters. 😭 

The name of this class needs to reflect that this is the query for a repository.  I think I'd suggest the following change (or something like it - if you have some other design you feel is better, I'm open to it). 

This isn't needed for pull requests, because they can only be queried [in relation to a repo](https://docs.github.com/en/rest/reference/pulls#list-pull-requests).

```suggestion
public abstract class GHIssueQueryBuilder extends GHQueryBuilder<GHIssue> {
    private final List<String> labels = new ArrayList<>();

    GHIssueQueryBuilder(GitHub root) {
        super(root);
    }

    // common query parameters such as "labels", "since", "state", "direction", etc., go in the parent class

    public static class ForRepository extends GHIssueQueryBuilder {
        private final GHRepository repo;

        public ForRepository(GHRepository repo) {
            super(repo.root());
            this.repo = repo;
        }
        
        @Override
        protected abstract String getApiUrl() {
            return repo.getApiTailUrl("issues");
        }
        // repo-specific parameters such as "milestone" and "assignee" go inside this class
    }
        
    // other query builders can be implemented later
   
}
```